### ISSUE TITLE
Complex Java Apps are impossible to use

### ISSUE THREAD

**Issue by ferrao** [baskerville/bspwm#1274]
I am really struggling with complex modern Java Apps that use Swing, JavaFX, HTML components, dialogs, etc.

Most problems manifest themselves by windows that either close immediately, or that are positioned on the middle of the screen and close as soon as cursor is moved and loose focus. Have tried all of the fixes that I could find online, targeting specific windows with rules, etc., but I never get to a solution that makes the whole app usable.

Most of these apps come with a splash window, followed by a login window, then the main window, which in turn  can trigger many other dialogs/pop-ups/etc.

If I launch the whole app as unmanaged, it will act as sticky across all desktops, which is not a solution.

In my mind, the proper fix would be to have all all dialogs/pop-ups/etc. not managed by bspwm at all, but I do not seem to find a way to get it done. Some of these windows close immediately or have the same class name as the main window, which makes me hard to target them with bspc rules.

Some help troubleshooting this by someone with a better knowledge on xwindows/bspwm would be really appreciated, I truly enjoy using bspwm and I am hating the prospect of leaving it because of this.

**Comment by emanuele6** [comment]
I recently had a similar problem with intellij idea; using [`wmname`](https://tools.suckless.org/x/wmname/) to set the wmname of the root window to `LG3D`.

    wmname LG3D

Running that command, fixed all problems immediately.

**Comment by ferrao** [comment]
I have tried all the possible solutions I could find @emanuele6, that one included ;)

**Comment by ferrao** [comment]
After quite a bit of work, I managed to setup my rules in such a way that I got all dialogs fixed on the center of the screen, which in turn allowed me to run xprops to get their proper classnames and then finally apply the proper rule, which in my case was:

`bspc rule -a install4j-jclient-Launcher:sun-awt-X11-XWindowPeer manage=off`

This managed to get 1 of my Java apps working fine, but I still think the whole process of applying rules to windows is not easy, specially when the windows we want to target are not visible, or disappear quickly, which makes it hard to use `xprops`.  

Perhaps some better docs/examples on how to use and apply rules would be really helpful.

**Comment by dieggsy** [comment]
@ferrao yeah, i've got `bspc rule -a 'jetbrains-studio:sun-awt-X11-XWindowPeer' manage=off` in my bspwmrc, so maybe the general rule `bspc rule -a '*:sun-awt-X11-XWindowPeer' manage=off` is more encompassing. Something I've done in the past wrt disappearing windowst is e.g. https://github.com/baskerville/bspwm/issues/1255#issuecomment-774561110 which can sometimes solve headaches, since you can open the window and hove hover it before xprop grabs the mouse

**Comment by ferrao** [comment]
> since you can open the window and hove hover it before xprop grabs the mouse

Nice tip @dieggsy, I actually tough about doing something like that, but was just getting so frustrated with the overall issues I was having with Java apps on `bspwm` and the fact that I hooped into `dwm` and it just worked, that I was very very close to leave bspwm as my window manager.

But ties like this are hard to break once you get used to things ;)

**Comment by ferrao** [comment]
Despite founding a work around for most of my issues, I do not think this problem is fixed.

There is plenty of room on the documentation, etc., to make this way better, or having a mode like `dwm` where the main window is tilled and everything else after that is not managed. 

Does anybody now if such thing is possible to configure in `bspwm`?

**Comment by felimarod** [comment]
> I recently had a similar problem with intellij idea; I used [`wmname`](https://tools.suckless.org/x/wmname/) to set the wmname of the root window to `LG3D`.
> 
> ```
> wmname LG3D
> ```
> 
> Running that command, fixed all problems immediately.

I had the same problem, I believed that I had wmname installed but not really.
I installed, put the config and this solve the issue.

Thanks.
